apps/bridge/src/utils/hooks/useApproveContract.ts

```diff
@@ -6,27 +6,32 @@ const { publicRuntimeConfig } = getConfig();
 
 type UseApproveContractProps = {
   contractAddress?: Address;
+  spender: Address;
   approveAmount: string;
   decimals: number;
+  bridgeDirection: 'deposit' | 'withdraw';
 };
 
 export function useApproveContract({
   contractAddress,
+  spender,
   approveAmount,
   decimals,
+  bridgeDirection,
 }: UseApproveContractProps) {
   const approveAmountBN =
     approveAmount === '' || Number.isNaN(Number(approveAmount))
       ? parseUnits('0', decimals)
       : parseUnits(approveAmount, decimals);
   const { config: depositConfig } = usePrepareContractWrite({
     address: contractAddress,
-    // TODO: Replace with dynamic abi importer
     abi: erc20ABI,
     functionName: 'approve',
-    chainId: parseInt(publicRuntimeConfig.l1ChainID),
-    // TODO: Add Allowance selection components
-    args: [publicRuntimeConfig.l1BridgeProxyAddress, approveAmountBN],
+    chainId:
+      bridgeDirection === 'deposit'
```

Comment:
Are these supposed to be different? Otherwise why have a ternary here?

Reply:
yeah they are, `l1ChainID` if deposit (approving an ERC20 on L1) and `l2ChainID` if withdrawal (approving an ERC20 on L2). lmk if im misunderstanding what you're referring to though

Reply:
oh that's my bad, it does say `l1` and `l2` but I thought they were the same in this case >.<